test/unit/org/apache/cassandra/cql3/ast/Conditional.java

```diff
@@ -34,32 +41,171 @@ default AbstractType<?> type()
+            lhs.toCQL(sb, formatter);
+            sb.append(' ').append(kind.value).append(' ');
+            rhs.toCQL(sb, formatter);
+        }
+
+        @Override
+        public Stream<? extends Element> stream()
+        {
+            return Stream.of(lhs, rhs);
+        }
+
+        @Override
+        public Conditional visit(Visitor v)
+        {
+            var u = v.visit(this);
+            if (u != this) return u;
+            var lhs = this.lhs.visit(v);
+            var rhs = this.rhs.visit(v);
+            if (lhs == this.lhs && rhs == this.rhs) return this;
+            return new Where(kind, lhs, rhs);
+        }
+    }
+
+    class Between implements Conditional
+    {
+        public final Expression ref, start, end;
+
+        public Between(Expression ref, Expression start, Expression end)
+        {
+            this.ref = ref;
+            this.start = start;
+            this.end = end;
+        }
+
+        @Override
+        public void toCQL(StringBuilder sb, CQLFormatter formatter)
+        {
+            ref.toCQL(sb, formatter);
+            sb.append(" BETWEEN ");
+            start.toCQL(sb, formatter);
+            sb.append(" AND ");
+            end.toCQL(sb, formatter);
+        }
+
+        @Override
+        public Stream<? extends Element> stream()
+        {
+            return Stream.of(ref, start, end);
+        }
+
+        @Override
+        public Conditional visit(Visitor v)
+        {
+            var u = v.visit(this);
+            if (u != this) return u;
+            Expression ref = this.ref.visit(v);
+            Expression start = this.start.visit(v);
+            Expression end = this.end.visit(v);
+            if (ref == this.ref && start == this.start && end == this.end) return this;
+            return new Between(ref, start, end);
+        }
+    }
+
     class In implements Conditional
     {
-        public final ReferenceExpression symbol;
+        public final Expression symbol;
         public final List<Expression> expressions;
 
-        public In(ReferenceExpression symbol, List<Expression> expressions)
+        public In(Expression symbol, List<Expression> expressions)
         {
             Invariants.checkArgument(!expressions.isEmpty());
             this.symbol = symbol;
             this.expressions = expressions;
         }
 
         @Override
-        public void toCQL(StringBuilder sb, int indent)
+        public void toCQL(StringBuilder sb, CQLFormatter formatter)
         {
-            symbol.toCQL(sb, indent);
+            symbol.toCQL(sb, formatter);
             sb.append(" IN ");
             sb.append('(');
             for (Expression e : expressions)
             {
-                e.toCQL(sb, indent);
+                e.toCQL(sb, formatter);
                 sb.append(", ");
             }
             sb.setLength(sb.length() - 2); // ", "
             sb.append(')');
         }
+
+        @Override
+        public Stream<? extends Element> stream()
```

Comment:
bug found in testing... forgot to define the stream so the sub elements that contained `bind` would break... this method is used to find what bind values to send to the driver